src/propmachinery.jl

```diff
@@ -1,32 +1,45 @@
-import Base: start, next, done
-
 abstract QuPropagatorMethod
 
-immutable QuPropagator{QPM<:QuPropagatorMethod}
-    hamiltonian
-    init_state::QuVector
+immutable QuPropagator{QPM<:QuPropagatorMethod, QVM<:Union(QuBase.AbstractQuVector,QuBase.AbstractQuMatrix)}
+    eq::QuEquation
```

Comment:
@acroy I tried something like this  

```julia
immutable QuPropagator{QPM<:QuPropagatorMethod, QVM<:Union(QuBase.AbstractQuVector,QuBase.AbstractQuMatrix), QE<:Union(QuEquation,QuBase.AbstractQuMatrix)}
    eq::QE
    init_state::QVM
    tlist
    method::QPM
    QuPropagator(eq, init_state, tlist, method) = new(eq, init_state, tlist, method)
end

QuPropagator{QPM<:QuPropagatorMethod, QV<:QuBase.AbstractQuVector, QE<:QuSchrodingerEq}(eq::QE, init_state::QV, tlist, method::QPM) = QuPropagator{QPM,QV,QE}(eq, init_state, tlist, method)

QuPropagator{QPM<:QuPropagatorMethod, QV<:QuBase.AbstractQuVector, QE<:QuBase.AbstractQuMatrix}(param::QE, init_state::QV,  tlist, method::QPM) = QuPropagator{QPM,QV,QE}(QuSchrodingerEq(param),init_state, tlist, method)

QuPropagator{QPM<:QuPropagatorMethod, QM<:QuBase.AbstractQuMatrix, QE<:QuLiouvillevonNeumannEq}(eq::QE, init_state::QM, tlist, method::QPM) = QuPropagator{QPM,QM,QE}(eq, init_state, tlist, method)

QuPropagator{QPM<:QuPropagatorMethod, QM<:QuBase.AbstractQuMatrix, QE<:QuBase.AbstractQuMatrix}(param::QE, init_state::QM,  tlist, method::QPM) = QuPropagator{QPM,QM,QE}(QuLiouvillevonNeumannEq(liouvillian_op(param)),init_state, tlist, method)
```
I end up with a convert error 
```julia
ERROR: `convert` has no method matching convert(::Type{QuArray{FiniteBasis{Orthonormal},Float64,2,Array{Float64,2}}}, ::QuLiouvillevonNeumannEq{QuArray{FiniteBasis{Orthonormal},Complex{Float64},2,SparseMatrixCSC{Complex{Float64},Int64}}})
```

QE also needs to support two types `QuEquation` and `AbstractQuMatrix` as we need the constructs which take which take 2 and 4 in the above. Any thoughts on this would be helpful.